cpu/atmega_common/periph/timer.c

```diff
@@ -136,9 +137,36 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
         return -1;
     }
 
+    unsigned state = irq_disable();
     ctx[tim].dev->OCR[channel] = (uint16_t)value;
     *ctx[tim].flag &= ~(1 << (channel + OCF1A));
     *ctx[tim].mask |= (1 << (channel + OCIE1A));
+    irq_restore(state);
+
+    return 0;
+}
+
+int timer_set(tim_t tim, int channel, unsigned int timeout)
+{
+    if (channel >= TIMER_CHANNELS) {
+        return -1;
+    }
+
+    unsigned state = irq_disable();
+    unsigned absolute = ctx[tim].dev->CNT + timeout;
+    ctx[tim].dev->OCR[channel] = absolute;
+    uint8_t mask = 1 << (channel + OCIE1A);
+    *ctx[tim].mask |= mask;
+    if ((absolute - ctx[tim].dev->CNT) > timeout) {
+        /* Timer already expired. Trigger the interrupt now and loop until it
+         * is triggered.
+         */
+        while (!(*ctx[tim].flag & (1 << (OCF1A + channel)))) {
+            ctx[tim].dev->OCR[channel] = ctx[tim].dev->CNT;
+        }
```

Comment:
Now I wonder: Is that really what we want?

The timer will wrap around, so when using the 'raw' `timer_set()` I'd expect it not to do anything clever but just to set the alarm. It would then be triggerd after the timer has overflowed.

Unfortunately the API is not very clear what should happen here.

Reply:
nvm, I confused it with `timer_set_absolute`